include/gz/gui/Application.hh

```diff
@@ -37,188 +37,195 @@ namespace tinyxml2
-      /// * A path
-      /// * A vector of plugins in that path
-      public: std::vector<std::pair<std::string, std::vector<std::string>>>
-          PluginList();
-
-      /// \brief Remove plugin by name. The plugin is removed from the
-      /// application and its shared library unloaded if this was its last
-      /// instance.
-      /// \param[in] _pluginName Plugn instance's unique name. This is the
-      /// plugin card's object name.
-      /// \return True if successful
-      public: bool RemovePlugin(const std::string &_pluginName);
-
-      /// \brief Get a plugin by its unique name.
-      /// \param[in] _pluginName Plugn instance's unique name. This is the
-      /// plugin card's object name.
-      /// \return Pointer to plugin object, null if not found.
-      public: std::shared_ptr<Plugin> PluginByName(
-          const std::string &_pluginName) const;
-
-      /// \brief Notify that a plugin has been added.
-      /// \param[in] _objectName Plugin's object name.
-      signals: void PluginAdded(const QString &_objectName);
-
-      /// \brief Callback when user requests to close a plugin
-      public slots: void OnPluginClose();
-
-      /// \brief Create a main window. Just calls InitializeMainWindow.
-      /// \return True if successful
-      /// \sa InitializeMainWindow
-      public: bool CreateMainWindow();
-
-      /// \brief Create a main window, populate with previously loaded plugins
-      /// and apply previously loaded configuration.
-      /// An empty window will be created if no plugins have been loaded.
-      /// \return True if successful
-      /// \sa LoadConfig
-      /// \sa LoadPlugin
-      private: bool InitializeMainWindow();
-
-      /// \brief Create individual dialogs for all previously loaded plugins.
-      /// This has no effect if no plugins have been loaded.
-      /// \return True if successful
-      /// \sa LoadConfig
-      /// \sa LoadPlugin
-      private: bool InitializeDialogs();
-
-      /// \brief Remove plugin by pointer.
-      /// \param[in] _plugin Shared pointer to plugin
-      private: void RemovePlugin(std::shared_ptr<Plugin> _plugin);
-
-      /// \brief Add previously loaded plugins to the main window.
-      /// \return True if successful. Will fail if the window hasn't been
-      /// created yet.
-      /// \sa LoadPlugin
-      private: bool AddPluginsToWindow();
-
-      /// \brief Apply previously loaded config to the main window.
-      /// \return True if successful, will fail if there's no main window
-      /// initialized.
-      private: bool ApplyConfig();
-
-      /// \internal
-      /// \brief Private data pointer
-      private: std::unique_ptr<ApplicationPrivate> dataPtr;
-    };
-
-    /// \brief Get current running application, this is a cast of qGuiApp.
-    /// \return Pointer to running application, or nullptr if none is running.
-    GZ_GUI_VISIBLE
-    Application *App();
-  }
-}
+class ApplicationPrivate;
```

Comment:
can we include these classes instead of this forward declaration ?

Reply:
I believe that at least `ApplicationPrivate` doesn't have a declaration in the public API, so we wouldn't be able to.